### PR TITLE
Add code to deploy composer in shared VPC environments

### PR DESCRIPTION
**Pull request by PradeepSingh1988**
## Pull Request Template
Please, go through these steps before you submit a PR.

1. Make sure that your PR is not a duplicate.
2. **Before** submitting the PR, review your changes:
    * Adjust the code to the existing [style](https://google.github.io/styleguide).
    * Add a set of unit tests for the changes and ensure they all pass.
    * Add accompanying README.md file with instructions on usage. See [awesome-readme](https://github.com/matiassingers/awesome-readme) for good examples of high-quality READMEs.
    * Add a link to your contribution in the top-level [README](https://github.com/GoogleCloudPlatform/professional-services/blob/master/README.md) (alpha-order).
    * Add [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0) license headers with an up-to-date copyright date attributed to `Google LLC`.
    * Remove unnecessary LICENSE files. There's no need to include an additional license since all repository submissions are covered by the top-level Apache 2.0 [license](https://github.com/GoogleCloudPlatform/professional-services/blob/master/LICENSE).
    * **For new tools/examples**, file your project with the PSO Engineering Council.

3. **After** these steps, you're ready to open a pull request.
    * Give a descriptive title to your PR.
    * Provide a description of your changes.
    * Include the Engineering Council bug ID with your pull request.
    * Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such).
<br/>

**PLEASE REMOVE THIS TEMPLATE BEFORE SUBMITTING**

### CODE REVIEW

File: examples/composer-shared-vpc/backend.tf [hunk 1]
```diff
@@ -0,0 +1,22 @@
+/**
```
Comment (juliocc) [inline examples/composer-shared-vpc/backend.tf:1]: As it is this makes your initial apply fail. I'd rename this file backen.tf.sample or something similar

**Review by juliocc — CHANGES_REQUESTED** [review]

File: examples/composer-shared-vpc/composer_v1_pvt_shared_vpc/variables.tf [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+  default     = true
+}
+
+variable "disk_size" {
+  description = "The disk size for nodes."
+  type        = string
+  default     = "100"
+}
+
+variable "oauth_scopes" {
```
Comment (juliocc) [inline examples/composer-shared-vpc/composer_v1_pvt_shared_vpc/variables.tf:90]: This variable is not used

File: examples/composer-shared-vpc/provider.tf [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  target_service_account = local.tf_sa
+  scopes                 = ["userinfo-email", "cloud-platform"]
+  lifetime               = "3600s"
+}
+
+/******************************************
+  Provider credential configuration
+ *****************************************/
+provider "google" {
+  access_token = data.google_service_account_access_token.default.access_token
```
Comment (juliocc) [inline examples/composer-shared-vpc/provider.tf:41]: Why don't you use [`impersonate_service_account`](https://registry.terraform.io/providers/hashicorp/google/latest/docs/guides/provider_reference#impersonate_service_account)

File: examples/composer-shared-vpc/terraform.tfvars [hunk 4]
```diff
@@ -0,0 +1,37 @@
+org_id                    = ""
```
Comment (juliocc) [inline examples/composer-shared-vpc/terraform.tfvars:1]: I'd rename this file with a .sample at the end

**Review by juliocc — CHANGES_REQUESTED** [review]

File: examples/composer-shared-vpc/backend.tf (hunk 1 above)
Comment (PradeepSingh1988) [reply examples/composer-shared-vpc/backend.tf]: Thanks @juliocc for review. Done

File: examples/composer-shared-vpc/composer_v1_pvt_shared_vpc/variables.tf (hunk 2 above)
Comment (PradeepSingh1988) [reply examples/composer-shared-vpc/composer_v1_pvt_shared_vpc/variables.tf]: Nice catch. Deleted the variable.

File: examples/composer-shared-vpc/provider.tf (hunk 3 above)
Comment (PradeepSingh1988) [reply examples/composer-shared-vpc/provider.tf]: Yes, modified it to [impersonate_service_account](https://registry.terraform.io/providers/hashicorp/google/latest/docs/guides/provider_reference#impersonate_service_account)

File: examples/composer-shared-vpc/terraform.tfvars (hunk 4 above)
Comment (PradeepSingh1988) [reply examples/composer-shared-vpc/terraform.tfvars]: Done

File: examples/composer-shared-vpc/composer_v1_pvt_shared_vpc/main.tf [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+}
+
+data "google_compute_subnetwork" "subnetwork" {
+  name    = var.subnetwork
+  project = var.network_project_id
+  region  = var.subnetwork_region
+}
+
+
+module "composer-v1-private" {
```
Comment (lcaggio) [inline examples/composer-shared-vpc/composer_v1_pvt_shared_vpc/main.tf:45]: I would add support for `software_config` (useful to specify env variables) and variables to setup private configuration (example: cloud_sql_ipv4_cidr)

File: examples/composer-shared-vpc/composer_v1_pvt_shared_vpc/variables.tf [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+  type        = list(string)
+  default = [
+    "roles/composer.worker",
+    "roles/iam.serviceAccountUser",
+    "roles/logging.logWriter"
+  ]
+}
+
+variable "assign_robot_sa_permissions" {
+  description = "Assign permissions to GKE and Composer robot agent service accpunts for Shared VPC"
```
Comment (lcaggio) [inline examples/composer-shared-vpc/composer_v1_pvt_shared_vpc/variables.tf:153]: Do we really need this flag? If we set it to false the Composer deployment will fail ...

**Comment by boredabdel** [comment]
@juliocc @lcaggio When this is ready to merge let me know ;)

**Comment by juliocc** [comment]
> @juliocc @lcaggio When this is ready to merge let me know ;)

My comments are resolved. I think there's a couple more from @lcaggio

File: examples/composer-shared-vpc/composer_v1_pvt_shared_vpc/main.tf (hunk 5 above)
Comment (PradeepSingh1988) [reply examples/composer-shared-vpc/composer_v1_pvt_shared_vpc/main.tf:45]: Done, added software config.

File: examples/composer-shared-vpc/composer_v1_pvt_shared_vpc/variables.tf (hunk 6 above)
Comment (PradeepSingh1988) [reply examples/composer-shared-vpc/composer_v1_pvt_shared_vpc/variables.tf]: Removed this config.

**Comment by lcaggio** [comment]
LGTM

**Review by juliocc — APPROVED** [review]

**Comment by PradeepSingh1988** [comment]
@boredabdel, @lcaggio and @juliocc have provided LGTM.

**Review by boredabdel — APPROVED** [review]
LGTM

**Pull request merged**